msprime/demography.py

```diff
@@ -1501,6 +1515,58 @@ def from_old_style(
                 population_map,
             )
 
+    @staticmethod
+    def from_tree_sequence(
+        ts: Union[tskit.TreeSequence, tskit.TableCollection]
+    ) -> Demography:
```

Comment:
I don't really want to start making *all* methods take tree sequences *or* table collections interchangeably - is there a particular reason to do it for this one? It's easy enough to do `from_tree_sequence(tables.tree_sequence())` if that's what you've got.

Reply:
Yep, that's a fair point. Since we need a TreeSequence here, it's probably a bad idea to go converting a set of tables into a tree sequence behind the scenes, since this can be expensive. Converting the other way around seems like a reasonable convenience in the ``initial_state`` argument to sim_ancestry though.